menu-api/Repositories/MenuItem_IngredientRepository.cs

```diff
@@ -15,29 +16,63 @@ public MenuItem_IngredientRepository(MenuContext context)
         //Add ingredient to menuItem
         public async Task AddIngredient(MenuItem_Ingredient menuItem_Ingredient)
         {
-            await context.MenuItem_Ingredients.AddAsync(menuItem_Ingredient);
-            await context.SaveChangesAsync();
+            var ingredient = await context.Ingredients.FindAsync(menuItem_Ingredient.IngredientId);
+            var menuItem = await context.MenuItems.FindAsync(menuItem_Ingredient.MenuItemId);
+
+            var _menuItem_Ingredient = await context.MenuItem_Ingredients.FindAsync(menuItem_Ingredient.MenuItemId, menuItem_Ingredient.IngredientId);
+
+            if (_menuItem_Ingredient == null && ingredient != null && menuItem !=null)
+            {
+                await context.MenuItem_Ingredients.AddAsync(menuItem_Ingredient);
+                await context.SaveChangesAsync();
+            }
+            else if (_menuItem_Ingredient != null)
+            {
+                throw new ItemAlreadyExsistsExeption();
+            }
+            else if (menuItem == null)
+            {
+                throw new ItemDoesNotExistExeption("MenuItem");
+            }
+            else if (ingredient == null)
+            {
+                throw new ItemDoesNotExistExeption("Ingredient");
+            }
         }
 
         //remove ingredient from menuItem
         public async Task RemoveIngredient(Guid menuItemId, Guid ingredientId)
         {
             MenuItem_Ingredient? menuItem_Ingredient = await context.MenuItem_Ingredients
                 .FindAsync(menuItemId, ingredientId);
+
             if (menuItem_Ingredient != null)
             {
                 context.MenuItem_Ingredients.Remove(menuItem_Ingredient);
                 await context.SaveChangesAsync();
             }
+            else
+            {
+                throw new ItemDoesNotExistExeption();
+            }
         }
 
         //remove all ingredients from menuItem
-        public async Task RemoveIngredients(Guid menuItemId)
+        public async Task RemoveAllIngredients(Guid menuItemId)
         {
-            IEnumerable<MenuItem_Ingredient> menuItem_Ingredients = context.MenuItem_Ingredients
+            var menuItem = await context.MenuItems.FindAsync(menuItemId);
+
+            if (menuItem != null)
```

Comment:
First check if the menu item does not exist, and then continue with the rest of the code